flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/SnapshotManager.java

```diff
@@ -211,16 +214,18 @@ public Optional<Snapshot> latestSnapshotOfUser(String user) {
         return findByListFiles(Math::min);
     }
 
-    public Long readHint(String fileName) {
+    public Long readHint(String fileName) throws IOException, InterruptedException{
         Path snapshotDir = snapshotDirectory();
         Path path = new Path(snapshotDir, fileName);
-        try {
+        int retryNumber = 0;
+        while (retryNumber++ < READ_HINT_RETRY_NUM) {
             if (path.getFileSystem().exists(path)) {
-                return Long.parseLong(FileUtils.readFileUtf8(path));
+                try {
+                    return Long.parseLong(FileUtils.readFileUtf8(path));
+                } catch (IOException ignored) {
+                }
             }
-        } catch (Exception e) {
-            LOG.info(
-                    "Failed to read hint file " + fileName + ". Falling back to listing files.", e);
+            TimeUnit.MILLISECONDS.sleep(READ_HINT_RETRY_INTERVAL);
```

Comment:
Catch and ignore the `InterruptedException`

Reply:
Done.